mesonbuild/build.py

```diff
@@ -2262,6 +2265,41 @@ def __init__(self, name, subdir, subproject, for_machine: MachineChoice, sources
     def get_default_install_dir(self, environment) -> T.Tuple[str, str]:
         return environment.get_shared_module_dir(), '{moduledir_shared}'
 
+class CompiledObjects(BuildTarget):
+    known_kwargs = known_objs_kwargs
+
+    def __init__(self, name, subdir, subproject, for_machine: MachineChoice, sources, objects, environment, kwargs):
+        self.typename = 'compiled objects'
+        if 'version' in kwargs:
+            raise MesonException('Built objects must not specify the version kwarg.')
```

Comment:
`InvalidArguments` is usually the specific exception used for this kind of problem